supportingFunctions.R

```diff
@@ -0,0 +1,118 @@
+#Elias Issa Rproject
+
+#Set Working Directory
+setwd("/Users/elias/Downloads/Rproject")
+#Initialize ggplot2
+library(ggplot2)
+
+#Converts all .txt files in a directory to csv files
+convertDelimiter<- function(directory){
+    #Object with all .txt file names in given directory
+    filedir <- dir(path = directory, pattern = ".txt")
+    #Loops through all .txt files, reads the content to data and writes a .csv file with the same name, then removes the .txt file
+    for(i in filedir){
+      data <- read.table(paste(directory,i, sep ="/" ), header = T, sep = "")
+      write.csv(data, file = paste(directory, sub(".txt",".csv", i), sep ="/"), quote=F, row.names=F, sep = ",")
+      file.remove(paste(directory,i, sep ="/" ))
+      }
+    }
+#Compiles all .csv files in a single directory
+#The user has 3 options in NA choice, remove rows with NAs through removeNA, include NAs but be warned of their presence through warnNA, or include NAs without warning through ignoreNA 
+compileCSV<- function(directory, choice){
+  #Deletes the output file in the event that it already exists
+  file.remove(paste(directory, "allData.csv", sep = "/"))
+  #Used to define the first addition to a file which will be later changed to FALSE
+  first = TRUE
+  #Object with all .csv files in the directory
+  filedir <- dir(path = directory, pattern = ".csv")
+  if(choice == "warnNA"){ print("Warning this summary data may contain NAs")}
+  #Loops through all .csv files in the directory
+  for(i in filedir){
+    #Stores the data in .csv files as temporary value
+    data2 <- read.csv(paste(directory,i, sep = "/"))
+    #Remove NAs if choice is removeNA
+    if(choice=="removeNA"){ data <- na.omit(data2)}

```

Comment:
There should be a condition when the user choose not to type either of these. 
-0.25